source/server/overload_manager_impl.cc

```diff
@@ -374,6 +399,48 @@ bool OverloadManagerImpl::registerForAction(const std::string& action,
   return true;
 }
 
+bool OverloadManagerImpl::tryAllocateResource(OverloadReactiveResourceName resource_name,
+                                              uint64_t increment,
+                                              ReactiveResourceUpdateCallbacks& cb) {
```

Comment:
I'm having some trouble understanding how listeners would call this method.  Some comments for the class method and interface method could help a lot.

I assume that tryAllocateResource can be called from any thread and would most likely be called from the thread associated with cb.dispatcher().  Having to go through a post for every accept would make it difficult to implement other desirable limits like accepting at most N connections per iteration of the event loop since all the accepts would happen from a possibly infinite chain of post calls.

Reply:
@antoniovicente this is most likely due to my knowledge gap, the only way i came up with is by posting on thread dispatcher from main thread.